hedera-node/docs/design/services/token-service/airdrops/airdrop-transaction.md

```diff
@@ -0,0 +1,206 @@
+              - It is an existing one and has `max_auto_associations` set to 0 and the recipient is not associated to the token →<br/>
+                Тhe token transfer is interpreted as a pending and a `TokenPendingAirdrop` entry is added to the `TransactionRecord`. In addition, the airdrop `PendingAirdropId`/`PendingAirdropValue` entry is added to the `PendingAirdrop` VirtualMap in state
+              - It is an existing one and is associated to the token →<br/>
+                Тhe airdrop is handled as a regular crypto transfer and a `TokenTransferList` is added to the `TransactionRecord`
+              - It’s not existing and the AccountID is a public `ECDSA` key →<br/>
+                Тhe airdrop would first auto-create a new account with `maxAutoAssociations` field of -1 and directly transfer the token being airdropped to it. This would include a `TokenTransferList` and the alias of the newly created account in the `TransactionRecord`
+              - It’s not existing and the AccountID is a public `ED25519` key →<br/>
+                Тhe airdrop would first auto-create a new account with `maxAutoAssociations` field of -1 and directly transfer the token being airdropped to it. This would include a `TokenTransferList` and the alias of the newly created account in the `TransactionRecord`
+              - It’s not existing and the AccountID is an `evm_address` →<br/>
+                Тhe airdrop would first lazily-create a new account with `maxAutoAssociations` field of -1 and directly transfer the token being airdropped to it. This would include a `TokenTransferList` and the alias of the newly created account in the `TransactionRecord`
+    - Fee calculation logic
+        - assess all required fees for the airdrop from the payer of the transaction. This includes:
+            - airdrop fee (assessed always)
+            - first auto-renew fee for the token (assessed always)
+            - custom token fees covered by the sender (assessed always if present)
+            - crypto transfer fee (pre-paid always)
+            - token association fee (pre-paid if account is not associated to the token)
+            - auto account creation (assessed if the recipient does not exist and they are referred by their public `ECDSA` key or `evm_address`)
+- Interact with the pending airdrop state - if the recipient is not associated to the token, then the `TokenAirdropTransaction` responsibilities in term of interacting with the pending state would be to just add the airdrop information in the state
+- Add this new operation type to the `ThroughputLimits` throttle bucket group, so that it's included in the throttling mechanism
+
+## Acceptance Tests
+
+* Verify that an airdrop with a token and existing recipient with `maxAutoAssociations=-1` works correctly and the recipient receives the token directly
```

Comment:
Please add a test that checks the behavior when the feature flag is disabled.

Reply:
Added